src/Void.Core/Filesystem.fs

```diff
@@ -0,0 +1,33 @@
+﻿namespace Void.Core
+
+module Filesystem =
+    open System
+    open System.IO
+
+    type LinesOrFailure =
+        | Lines of string seq
+        | Failure of Error
+
+    let private readLines path =
+        try
+            File.ReadLines path |> Lines
+        with
+        | :? UnauthorizedAccessException ->

```

Comment:
Interesting, I hadn't though it of like that before!

`Exception e` is kind of like your catch-all case, huh?